main/src/main/java/org/mobilitydata/gtfsvalidator/validator/DateTripsValidator.java

```diff
@@ -0,0 +1,109 @@
+import org.mobilitydata.gtfsvalidator.table.*;
+import org.mobilitydata.gtfsvalidator.type.GtfsDate;
+import org.mobilitydata.gtfsvalidator.util.CalendarUtil;
+import org.mobilitydata.gtfsvalidator.util.TripCalendarUtil;
+
+/**
+ * Validates that trip data exists for the next 7 days. Dates with the majority trips per day should
+ * be included for at least the next 7 day period. see
+ * https://github.com/MobilityData/gtfs-validator/issues/886#issuecomment-832237225
+ *
+ * <p>Generated notices:
+ *
+ * <ul>
+ *   <li>{@link TripDataShouldBeValidForNext7DaysNotice}.
+ * </ul>
+ */
+@GtfsValidator
+public class DateTripsValidator extends FileValidator {
+
+  private final GtfsCalendarDateTableContainer calendarDateTable;
+  private final GtfsCalendarTableContainer calendarTable;
+  private final GtfsTripTableContainer tripContainer;
+  private final GtfsFrequencyTableContainer frequencyTable;
+  private final CurrentDateTime currentDateTime;
+
+  @Inject
+  DateTripsValidator(
+      CurrentDateTime currentDateTime,
+      GtfsCalendarDateTableContainer calendarDateTable,
+      GtfsCalendarTableContainer calendarTable,
+      GtfsTripTableContainer tripContainer,
+      GtfsFrequencyTableContainer frequencyTable) {
+    this.currentDateTime = currentDateTime;
+    this.calendarTable = calendarTable;
+    this.calendarDateTable = calendarDateTable;
+    this.tripContainer = tripContainer;
+    this.frequencyTable = frequencyTable;
+  }
+
+  @Override
+  public void validate(NoticeContainer noticeContainer) {
+
+    LocalDate now = currentDateTime.getNow().toLocalDate();
+
+    final Map<String, SortedSet<LocalDate>> servicePeriodMap =
+        CalendarUtil.servicePeriodToServiceDatesMap(
+            CalendarUtil.buildServicePeriodMap(calendarTable, calendarDateTable));
+
+    var tripCounts =
+        TripCalendarUtil.countTripsForEachServiceDate(
+            servicePeriodMap, tripContainer, frequencyTable);
+    var majorityServiceDates = TripCalendarUtil.computeMajorityServiceCoverage(tripCounts);
+    LocalDate currentDatePlusSevenDays = now.plusDays(7);
+
+    if (majorityServiceDates != null) {
+      LocalDate serviceWindowStartDate = majorityServiceDates.get().startDate();
+      LocalDate serviceWindowEndDate = majorityServiceDates.get().endDate();
+      if (serviceWindowStartDate.isAfter(now)
+          || serviceWindowEndDate.isBefore(currentDatePlusSevenDays)) {
+        noticeContainer.addValidationNotice(
+            new TripDataShouldBeValidForNext7DaysNotice(
+                GtfsDate.fromLocalDate(now),
+                GtfsDate.fromLocalDate(serviceWindowStartDate),
+                GtfsDate.fromLocalDate(serviceWindowEndDate)));
+      }
+    }
+  }
+
+  static class TripDataShouldBeValidForNext7DaysNotice extends ValidationNotice {
+    private final GtfsDate currentDate;
+    private final GtfsDate serviceWindowStartDate;
+    private final GtfsDate serviceWindowEndDate;
```

Comment:
Could you add a comment to each field here to match the comments in RULES.md?  It would make my life easier regarding PR #1346.